File: src/main/java/net/exotia/exotiaprofanity/client/ProfanityClient.java
package net.exotia.exotiaprofanity.client;

import net.exotia.exotiaprofanity.configuration.Configuration;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.plugin.Plugin;
import org.java_websocket.client.WebSocketClient;
import org.java_websocket.handshake.ServerHandshake;
import java.util.UUID;
import java.util.logging.Logger;

public class ProfanityClient extends WebSocketClient {
    private final Plugin plugin;
    private final Logger logger;
    private final Configuration configuration;

    public ProfanityClient(Configuration configuration, Plugin plugin) {
        super(configuration.getApiUrl());
        this.plugin = plugin;
        this.logger = plugin.getLogger();
        this.configuration = configuration;
    }

    public Configuration getConfiguration() {
        return this.configuration;
    }

    @Override
    public void onOpen(ServerHandshake serverHandshake) {
        this.logger.info("New connection has been opened!");
    }
    @Override
    public void onMessage(String message) {
        String[] parts = message.split(this.configuration.getDelimiter());
        if (parts.length < 2) return;

        this.plugin.getServer().getScheduler().runTask(this.plugin, () -> {
            Player player = Bukkit.getPlayer(UUID.fromString(parts[0]));
            if (player == null) return;
            player.chat(parts[1]);
        });
    }
    @Override
    public void onClose(int code, String reason, boolean remote) {
        this.logger.severe("Closed with exit code " + code + " additional info: " + reason);
        if (this.configuration.isAutoReconnect()) {
            this.logger.info("Attempting to reconnect...");
            this.plugin.getServer().getScheduler().runTask(this.plugin, this::connect);
        }
    }
    @Override
    public void onError(Exception exception) {
        this.logger.severe("An error occurred:" + exception);
    }
}

File: src/main/java/net/exotia/exotiaprofanitykotlin/client/ProfanityClient.kt
package net.exotia.exotiaprofanitykotlin.client

import net.exotia.exotiaprofanitykotlin.configuration.Configuration
import net.exotia.exotiaprofanitykotlin.ExotiaProfanityKotlin
import okhttp3.Response
import okhttp3.WebSocket
import okhttp3.WebSocketListener
import okhttp3.OkHttpClient
import okhttp3.Request
import org.bukkit.Bukkit
import org.bukkit.plugin.Plugin
import java.util.logging.Logger
import java.util.UUID
import kotlin.reflect.KFunction


class ProfanityClient(
  private val plugin: Plugin,
  private val logger: Logger,
  private val configuration: Configuration,
  private var tryToReconnect: () -> Unit
) : WebSocketListener() {

  override fun onOpen(webSocket: WebSocket, response: Response) {
    this.logger.info("New connection has been opened!")
  }

  override fun onMessage(webSocket: WebSocket, text: String) {
    val parts: List<String> = text.split(this.configuration.delimiter)
    if (parts.size < 2) return

    this.plugin.server.scheduler.runTask(this.plugin, Runnable {
      val player = Bukkit.getPlayer(UUID.fromString(parts[0]))
      player?.chat(parts[1])
    })
  }

  override fun onClosed(webSocket: WebSocket, code: Int, reason: String) {
    this.logger.severe("Closed with exit code $code additional info: $reason");
    webSocket.close(code, reason)
    this.tryToReconnect()
  }

  override fun onFailure(webSocket: WebSocket, t: Throwable, response: Response?) {
    this.logger.severe("WebSocket failure!")
    if (t.message != null) this.logger.info(t.message)
    this.tryToReconnect()
  }
}
